rocksdb_admin/application_db_backup_manager.cpp

```diff
@@ -185,17 +210,90 @@ bool ApplicationDBBackupManager::backupDBToS3(const std::shared_ptr<ApplicationD
   std::vector<std::string> checkpoint_files;
   status = rocksdb::Env::Default()->GetChildren(checkpoint_local_path, &checkpoint_files);
   if (!status.ok()) {
-    // OKOrSetException(status, AdminErrorCode::DB_ADMIN_ERROR, &callback);
     LOG(ERROR) << "Error happened when trying to list files in the checkpoint: " << status.ToString();
     common::Stats::get()->Incr(kS3BackupFailure);
     return false;
   }
 
-  // Upload checkpoint to s3
+  // find the timestamp of last backup
+  int64_t last_ts = -1;
+  auto itor = db_backups_.find(db->db_name());
+  if (itor != db_backups_.end()) {
+    last_ts = itor->second[(int)itor->second.size()-1];
+  }
+
+  // if there exists last backup, we need to get the backup_descriptor
   auto local_s3_util = createLocalS3Util(FLAGS_incre_backup_limit_mbs, FLAGS_s3_incre_backup_bucket);
-  std::string formatted_s3_dir_path = folly::stringPrintf("%s%s/%d/", 
+  std::string formatted_s3_dir_path_upload = folly::stringPrintf("%s%s/%ld/", 
     ensure_ends_with_pathsep(FLAGS_s3_incre_backup_prefix).c_str(), db->db_name().c_str(), ts);
   std::string formatted_checkpoint_local_path = ensure_ends_with_pathsep(checkpoint_local_path);
+
+  std::unordered_map<std::string, int64_t> file_to_ts;
+  const string sst_suffix = ".sst";
+
+  if (last_ts >= 0) {
```

Comment:
In this case if machine restarts we wouldnt have last_ts.

Can you leave a comment that we need to handle this code doesn't handle this case well?

Thanks!